### PR TITLE
update pipeline presubmit test

### PR DESCRIPTION
**Pull request by IronPan**
Deprecate e2e test in favor of the e2e test v2
Rename the old sample test to pending deprecate

### CODE REVIEW

**Comment by IronPan** [comment]
cc @gaoning777

File: config/jobs/kubeflow/kubeflow-presubmits.yaml [hunk 1]
```diff
@@ -245,7 +229,7 @@ presubmits:
           - e2e_test_gke_v2.yaml
           - --test_result_folder
           - e2e_test
-    - name: presubmit-sample-test-v2
+    - name: presubmit-sample-test
```
Comment (jlewi) [inline config/jobs/kubeflow/kubeflow-presubmits.yaml]: Can you follow the naming convention in this file "<repo>-<test descriptor>".
This makes it easier to identify tests in prow jobs dashboard and other places.
https://prow.k8s.io/?repo=kubeflow%2Fkubeflow

Comment (IronPan) [reply config/jobs/kubeflow/kubeflow-presubmits.yaml]: done

File: config/jobs/kubeflow/kubeflow-presubmits.yaml [hunk 2]
```diff
@@ -214,25 +214,9 @@ presubmits:
         - --workflow_file
         - build_image.yaml
     run_after_success:
-    - name: presubmit-e2e-test
+    - name: e2e-test-presubmit
```
Comment (jlewi) [inline config/jobs/kubeflow/kubeflow-presubmits.yaml]: The convention is to start the name of ever test with the repo it belongs to. Can you change the name please?

Comment (BenTheElder) [reply config/jobs/kubeflow/kubeflow-presubmits.yaml]: please follow the convention, the CI jobs in particular live in a flat namespace for logs (thankfully presubmit do not)

Comment (IronPan) [reply config/jobs/kubeflow/kubeflow-presubmits.yaml]: done

**Comment by IronPan** [comment]
comment addressed. PTAL

**Comment by krzyzacy** [comment]
/lgtm
/hold
@jlewi lgty?

**Pull request merged**